src/Map/ProspectorOverlayLayer.cs

```diff
@@ -57,12 +57,10 @@ public ProspectorOverlayLayer(ICoreAPI api, IWorldMapManager mapSink) : base(api
                         invMan.SlotModified += Event_SlotModified;
                     }
                 };
-                _clientApi.Event.PlayerLeave += (p) =>
-                {
-                    if (p == _clientApi?.World.Player)
+                _clientApi.Event.LeaveWorld += () => {
```

Comment:
Are you sure we don't need the event cleanup here anymore ?
Also should saving not also happen on a regular basis (like on a timer) and reset the HasChanged-Flag ?

Reply:
From my observations while debugging (at least with 1.17.11), I noticed that this event was never fired. I guess a player is not informed about himself leaving? So the cleanup code was never executed anyway. I'm not sure about the lifecycle of the client api object, as we also don't have any cleanup for `Event.ChatMessage` or `Event.AfterActiveSlotChanged`.